### PR TITLE
Add the ability to add a ⭐ to a message in the #starboard channel as well as the original message

### PR DESCRIPTION
**Pull request by JGDeveloper06**
Issue : [Ability to add a ⭐from Starboard and Original Message](https://github.com/Java-Discord/JavaBot/issues/20)
Its notable that we cannot ⭐ a message directly from the starboard. 

I have added the feature to vote on the message both from the starboard and the original message. With this, I have implemented it in a way that it counts a User only once even if the User votes on both the messages.

It closes the issue : [Ability to add a ⭐from Starboard and Original Message](https://github.com/Java-Discord/JavaBot/issues/20)

### CODE REVIEW

File: src/main/java/net/javadiscord/javabot/events/StarboardListener.java [hunk 1]
```diff
@@ -190,21 +209,55 @@ public void onMessageReactionRemove(@NotNull MessageReactionRemoveEvent event) {
...
+		}
 	}
 
 	@Override
 	public void onMessageDelete(@NotNull MessageDeleteEvent event) {
 		removeMessageFromStarboard(event.getGuild(), event.getMessageId());
 	}
+
+	private void updateSbReactionCounts(GenericMessageReactionEvent event, StarBoardConfig config, String messageId, String remoteTextChannelId) {
+		Objects.requireNonNull(event.getGuild().getTextChannelById(remoteTextChannelId)).retrieveMessageById(messageId).queue(message -> {
```
Comment (sahilasopa) [inline src/main/java/net/javadiscord/javabot/events/StarboardListener.java:237]: replace `Objects.requireNonNull` with a null check `Objects.requireNonNull` throws exceptions

File: src/main/java/net/javadiscord/javabot/events/StarboardListener.java [hunk 2]
```diff
@@ -190,21 +209,55 @@ public void onMessageReactionRemove(@NotNull MessageReactionRemoveEvent event) {
...
+							prevChannelId,
+							this.getUniqueStarCounts(message, msg, config)-1,
+							message, config);
+			});
+		});
+	}
+
+	private int getUniqueStarCounts(Message starboardMessage, Message originalMessage, StarBoardConfig config){
+		try {
+			List<User> list = originalMessage.retrieveReactionUsers(config.getEmotes().get(0)).takeAsync(this.getReactionCountForEmote(config.getEmotes().get(0), originalMessage)).get();
```
Comment (sahilasopa) [inline src/main/java/net/javadiscord/javabot/events/StarboardListener.java:255]: give these variables a valid name instead so list

File: src/main/java/net/javadiscord/javabot/events/StarboardListener.java [hunk 3]
```diff
@@ -190,21 +209,55 @@ public void onMessageReactionRemove(@NotNull MessageReactionRemoveEvent event) {
...
+			List<User> list2 = starboardMessage.retrieveReactionUsers(config.getEmotes().get(0)).takeAsync(this.getReactionCountForEmote(config.getEmotes().get(0), starboardMessage)).get();
```
Comment (sahilasopa) [inline src/main/java/net/javadiscord/javabot/events/StarboardListener.java:256]: same here

File: src/main/java/net/javadiscord/javabot/events/StarboardListener.java [hunk 4]
```diff
@@ -190,21 +209,55 @@ public void onMessageReactionRemove(@NotNull MessageReactionRemoveEvent event) {
...
 		removeMessageFromStarboard(event.getGuild(), event.getMessageId());
 	}
+
+	private void updateSbReactionCounts(GenericMessageReactionEvent event, StarBoardConfig config, String messageId, String remoteTextChannelId) {
+		Objects.requireNonNull(event.getGuild().getTextChannelById(remoteTextChannelId)).retrieveMessageById(messageId).queue(message -> {
+			MessageEmbed me = message.getEmbeds().get(0);
+			String prevChannelId = Objects.requireNonNull(me.getAuthor().getUrl()).split("/")[5];
+			String originalMessageId = me.getAuthor().getUrl().split("/")[6];
+
+			Objects.requireNonNull(event.getGuild().getTextChannelById(prevChannelId)).retrieveMessageById(originalMessageId).queue(msg -> {
```
Comment (sahilasopa) [inline src/main/java/net/javadiscord/javabot/events/StarboardListener.java:242]: replace Objects.requireNonNull with a null check here aswell

**Review by sahilasopa — CHANGES_REQUESTED** [review]

**Comment by DynxstyGIT** [comment]
I'm sorry to say that, but I have to close this PR, as we're going to refactor the whole starboard to use the H2 Database.

**Pull request closed without merging**